src/server/request.rs

```diff
@@ -66,17 +68,31 @@ impl<'a, T> Request<'a, T> {
     #[inline]
     pub fn version(&self) -> &HttpVersion { &self.version }
 
-    /*
     /// The target path of this Request.
     #[inline]
-    pub fn path(&self) -> Option<&str> {
-        match *self.uri {
-            RequestUri::AbsolutePath(ref s) => Some(s),
-            RequestUri::AbsoluteUri(ref url) => Some(&url[::url::Position::BeforePath..]),
-            _ => None
+    pub fn path(&self) -> Option<Cow<str>> {
+        match self.uri {
+            RequestUri::AbsolutePath(ref s) => {
+                let url = Url::parse("http://example.com").ok().and_then(|u| u.join(s).ok());
```

Comment:
The [spec claims this variant has "absolute-path [ "?" query ]"](https://tools.ietf.org/html/rfc7230#section-5.3.1). Maybe its worth adjusting the variant to `AbsolutePath { path: String, query: Option<String> }`?